### PR TITLE
Bug 2075548: [DownstreamMerge] 09 Jan 2023

### PR DESCRIPTION
**Pull request by kyrtapz**
Modified https://github.com/openshift/ovn-kubernetes/pull/1466/commits/4af1bf20f783be9c48847bf5789288493239077d to remove LrpMAC from downstream tests.
Added https://github.com/openshift/ovn-kubernetes/pull/1466/commits/e74520379c7176ce99a24ed6cd93d8dd52983b7f to avoid data race in downstream hybrid tests:
```
WARNING: DATA RACE
Read at 0x00000442cb09 by goroutine 1638:
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*DefaultNetworkController).addLogicalPort()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/ovn/pods.go:192 +0xbfa
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*DefaultNetworkController).ensurePod()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/ovn/ovn.go:141 +0x9b5
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*defaultNetworkControllerEventHandler).AddResource()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/ovn/default_network_controller.go:622 +0x1a4
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry.(*RetryFramework).resourceRetry.func1()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/retry/obj_retry.go:323 +0x1cdd
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry.(*RetryFramework).DoWithLock()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/retry/obj_retry.go:109 +0x12a
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry.(*RetryFramework).resourceRetry()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/retry/obj_retry.go:223 +0xc4
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry.(*RetryFramework).iterateRetryResources.func1()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/retry/obj_retry.go:363 +0xea
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry.(*RetryFramework).iterateRetryResources.func2()
      /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/retry/obj_retry.go:364 +0x58

Previous write at 0x00000442cb08 by goroutine 157:
  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config.PrepareTestConfig()
```

### CODE REVIEW

**Comment by tssurya** [comment]
/retitle Bug 2075548: [DownstreamMerge] 09 Jan 2023

File: go-controller/pkg/ovn/hybrid_test.go [hunk 1]
```diff
@@ -1313,7 +1313,7 @@ var _ = ginkgo.Describe("Hybrid SDN Master Operations", func() {
 			err = f.Start()
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 			clusterController := NewOvnController(fakeClient, f, stopChan, addressset.NewFakeAddressSetFactory(),
-				libovsdbOvnNBClient, libovsdbOvnSBClient, record.NewFakeRecorder(10), &sync.WaitGroup{})
+				libovsdbOvnNBClient, libovsdbOvnSBClient, record.NewFakeRecorder(10), wg)
```
Comment (flavio-fernandes) [inline go-controller/pkg/ovn/hybrid_test.go:1316]: NIT: do we need a CARRY and or DOWNSTREAM HACK comments in the code to know this is DS only?

Comment (kyrtapz) [reply go-controller/pkg/ovn/hybrid_test.go:1316]: these two tests code are not there in upstream :)

Comment (flavio-fernandes) [reply go-controller/pkg/ovn/hybrid_test.go:1316]: right, which may mean we should have a carry or something similar flag for us to remember that. Let me find an example

**Pull request merged**